File: java-practice/src/main/java/com/doctor/javascript/ebook/the_definitive_guide6/Chapter8Code.java
package com.doctor.javascript.ebook.the_definitive_guide6;

import com.doctor.javascript.JavaForJsUtil;
import javax.script.ScriptEngine;
import java.io.Reader;
/**

 * Created by doctor on 15-2-14.
 */
public class Chapter8Code {
    public static  void main(String[] args) throws  Throwable{
        ScriptEngine engine = JavaForJsUtil.getNashornScriptEngine();
        Reader reader = JavaForJsUtil.getDefaultJSFileInPackage(Chapter8Code.class);
        engine.eval(reader);
    }
}

File: java-practice/src/main/java/com/doctor/javascript/ebook/the_definitive_guide6/Chapter8Code.js
/**
 * Created by doctor on 15-2-14.
 */

var calculator = {
    oper1:1,
    oper2:5,
    add:function() {
        this.results = this.oper1+ this.oper2;
    }
}
calculator.add();
print(calculator.results);